packages/api/src/platforms/vtex/clients/commerce/index.ts

```diff
@@ -201,49 +202,51 @@ export const VtexCommerce = (
             }
           : null
 
+        const addressSession = body?.selectedAddresses?.map((address) => {
+          const addressSession: SelectedAddress = {
+            addressType: address.addressType || null,
+            receiverName: address.receiverName || null,
+            postalCode:
+              address.postalCode || incrementedAddress?.postalCode || null,
+            city: incrementedAddress?.city || null,
+            state: incrementedAddress?.state || null,
+            country: address.country || incrementedAddress?.country || null,
+            street: incrementedAddress?.street || null,
+            number: incrementedAddress?.number || null,
+            neighborhood: incrementedAddress?.neighborhood || null,
+            complement: incrementedAddress?.complement || null,
+            reference: incrementedAddress?.reference || null,
+            geoCoordinates: [], // Initialize with default value
+          }
+
+          const longitude =
+            address?.geoCoordinates instanceof Array
+              ? null
```

Comment:
if it is an instance of array, shouldn't we look for `address?.geoCoordinates[0]` for the longitude?

Reply:
I think its correct because it´s { latitude: 46.3, longitude: 0.35 } so I´m turning into 46.3 and 0.35 and creating my array at the checkout orderForm here
 addressSession.geoCoordinates =
            longitude && latitude
              ? [longitude, latitude]
              : incrementedAddress?.geoCoordinates || []

Reply:
I see, then why do we check if it is an array?

Reply:
Changed :)